test/index.spec.ts

```diff
@@ -602,4 +603,33 @@ describe("MockAxios", () => {
             expect(() => cancelToken.token.throwIfRequested()).toThrow();
         });
     });
+
+    describe("requestHandler", () => {
+        it("provides a method to set up a requestHandler", () => {
+            expect(MockAxios).toHaveProperty("requestHandler");
+        });
+
+        it("invokes the requestHandler on every incoming request", () => {
+            const requestHandler = jest.fn();
+            MockAxios.requestHandler(requestHandler);
+
+            MockAxios.get();
+            MockAxios.post();
+
+            expect(requestHandler).toHaveBeenCalledTimes(2);
+        });
+
+        it("supplies the latest queue item to the requestHandler", () => {
+            let lastRequestHanderRequest: AxiosMockQueueItem;
+
+            MockAxios.requestHandler((req: AxiosMockQueueItem) => {
+                lastRequestHanderRequest = req;
+            });
+
+            MockAxios.post();
+
+            const lastPromise: AxiosMockQueueItem = MockAxios.post();
```

Comment:
The type here is actually ``UnresolvedSynchronousPromise<any>` but I think we can leave it out completely.

```suggestion
            const lastPromise = MockAxios.post();
```